<!doctype html>
<html lang="en">

	<head>
		<meta charset="utf-8">

		<title>Medium data</title>

		<meta name="author" content="Trevor Bedford">

		<meta name="apple-mobile-web-app-capable" content="yes" />
		<meta name="apple-mobile-web-app-status-bar-style" content="black-translucent" />

		<meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no">

		<link rel="stylesheet" href="css/reveal.min.css">
		<link rel="stylesheet" href="css/theme/trvrb.css" id="theme">

		<!-- For syntax highlighting -->
		<link rel="stylesheet" href="lib/css/zenburn.css">

		<!-- If the query includes 'print-pdf', include the PDF print sheet -->
		<script>
			if( window.location.search.match( /print-pdf/gi ) ) {
				var link = document.createElement( 'link' );
				link.rel = 'stylesheet';
				link.type = 'text/css';
				link.href = 'css/print/pdf.css';
				document.getElementsByTagName( 'head' )[0].appendChild( link );
			}
		</script>

		<!--[if lt IE 9]>
		<script src="lib/js/html5shiv.js"></script>
		<![endif]-->
	</head>

	<body>

		<div style="position: absolute; top:10px; left:10px; z-index:100;">
			<a href="http://bedford.io">
				<img width="20px" src="images/logo.png">
			</a>
		</div>

		<div class="reveal">

			<!-- Any section element inside of this container is displayed as a slide -->
			<div class="slides">
				
				<section>
					<h1><span class="fragment strike-through">Big data</span></h1>
					<br>
					<p><a href="http://bedford.io">Trevor Bedford</a> (<a href="http://twitter.com/trvrb">@trvrb</a>)
					<p><a href="http://fhcrc.org">FHCRC</a>
					<p>Aug 1, 2014
				</section>	
				
				<section>
					<h2>Medium data</h2>
					<br>
					<p><a href="http://bedford.io">Trevor Bedford</a> (<a href="http://twitter.com/trvrb">@trvrb</a>)
					<p><a href="http://fhcrc.org">FHCRC</a>
					<p>Aug 1, 2014
				</section>
				
				<section>
					Slides at <br> <a href="http://bedford.io">http://bedford.io/projects/medium-data/</a>
				</section>
				
				<section>
					<h2>Data tells a story</h2>
				</section>						
				
				<section>
					<h3><a href="http://en.wikipedia.org/wiki/Charles_Joseph_Minard">Charles Minard</a> and Napoleon's 1812 campaign</h3>
					<a class="stretch" href="images/minard.png"><img src="images/minard.png"></a>
				</section>				
	
				<section>
					<h2>The <i>EPAS1</i> gene and high-altitude adaptation in Tibetans</h2>
				</section>		
				
				<section>
					<h3>Tibet is high</h3>
					<img class="stretch" src="images/tibet.jpg">
				</section>		
				
				<section>
					<h3>Tibetans are adapted to high altitudes</h3>
					<ul>
					<li>~600,000 people live at 14,800+ ft in elevation, where oxygen is only 60% of sea level</li>
					<li>Among other traits, Tibetans are protected from <a href="http://en.wikipedia.org/wiki/Chronic_mountain_sickness">chronic mountain sickness</a>, exhibiting little increase in hemoglobin concentration at high altitudes</li>
					</ul>
				</section>									

				<section>
					<h3><a href="http://www.pnas.org/content/107/25/11459.abstract">Tibetan <i>EPAS1</i> is diverged from other populations</a></h3>
					<img class="stretch" src="images/epas1_manhattan_plot.png">
				</section>
				
				<section>
					<h3><a href="http://www.sciencemag.org/content/329/5987/75.abstract">Tibetan <i>EPAS1</i> is diverged from other populations</a></h3>
					<img class="stretch" src="images/epas1_heatmap.jpg">
				</section>	
				
				<section>
					<h2>EPAS1 is a transcription factor involved in the body's response to oxygen level</h2>
				</section>	
				
				<section>
					<h2>Story was <i>in situ</i> adaptation in Tibetans to high altitude</h2>
				</section>														
				
				<section>
					<h3><a href="http://www.nature.com/nature/journal/vaop/ncurrent/full/nature13408.html">Tibetan <i>EPAS1</i> derives from Denisovan DNA</a></h3>
					<img class="stretch" src="images/epas1_haplotype_plot.png">
				</section>	
				
				<section>
					<h2>Denisovans?</h2>
				</section>					
				
				<section data-background="images/denisovan_family.jpg">
				</section>	

				<section data-background="images/denisovan_cave.jpg">
				</section>	

				<section data-background="images/denisovan_excavation.jpg">
				</section>	

				<section data-background="images/denisovan_fingerbone.jpg">
				</section>	

				<section data-background="images/denisovan_sequencing.jpg">
				</section>	
				
				<section data-background="images/dna_sequences.jpg">
				</section>					
				
				<section>
					<h3><a href="http://www.nature.com/nature/journal/v505/n7481/full/nature12886.html">Evolutionary relationships between human species</a></h3>
					<img class="stretch" src="images/denisovan_tree.jpg">
				</section>	
				
				<section>
					<h3><a href="http://www.nature.com/nature/journal/v505/n7481/full/nature12886.html">Extinction of Neanderthals and Denisovans</a></h3>
					<img class="stretch" src="images/denisovan_ne.jpg">
				</section>		
				
				<section data-background="#222222">
					<img class="stretch" src="images/human_migration.jpg">
				</section>	
				
				<section>
					<h3><a href="http://www.nature.com/nature/journal/v505/n7481/full/nature12886.html">Relatedness between modern and archaic populations</a></h3>
					<img class="stretch" src="images/denisovan_relatedness.jpg">
				</section>					

				<section data-background="#222222">
					<img class="stretch" src="images/out_of_africa.jpg">
				</section>	
									
				<section data-background="#222222">
					<img class="stretch" src="images/middle_earth.jpg">
				</section>																				
				
				<section>
					<h3>via <a href="http://xkcd.com/657/">XKCD</a></h3>
					<a class="stretch" href="images/xkcd_lord_of_the_rings.png"><img src="images/xkcd_lord_of_the_rings.png"></a>
				</section>	
				
				<section>
					<h2>Working with data</h2>
				</section>	
				
				<section>
					<h2>Step 1. Download data from public repository</h2>
					For example, <a href="http://fludb.org">fludb.org</a>
				</section>	
				
				<section>
					<h2>Step 2. Clean up data and get it into a proper format</h2>
					<pre><code data-trim>
strain	date	date_decimal	date_precision	region	country	general_location	specific_location	source	accession	sequence
A/Alabama/WRAIR1242P/2009	2009-02-01	2009.085	day	USACanada	USA	Alabama	Alabama	IRD	CY093263	AACGCTGTGCCTTGGGCACCATGCAGTACCAAACGGAACGATAGTGAAAACAATCACGAATGACCAAATTGAAGTTACTA
A/Alexandroupolis/99/2007	2007-02-15	2007.123	day	Europe	Greece	Greece	Alexandroupolis	IRD	EU716078	AACGCTGTGCCTTGGGCACCATGCAGTACCAAACGGAACGATAGTGAAAACAATCACGAATGACCAAATTGAAGTTACTA
A/Ashburton/280/2004	2004-10-12	2004.781	day	Oceania	NewZealand	NewZealand	Ashburton	IRD	CY002954	AACGCTGTGCCTTGGGCACCATGCAGTACCAAACGGAACGATAGTGAAAACAATCACGAATGACCAAATTGAAGTTACTA
A/Auckland/599/2000	2000-09-22	2000.726	day	Oceania	NewZealand	NewZealand	Auckland	IRD	CY023034	AACGCTGTGCCTGGGGCACCATGCAGTGCCAAACGGAACGCTAGTGAAAACAATCACGAATGACCAAATTGAAGTGACTA
A/Australia/34/2009	2009-07-24	2009.559	day	Oceania	Australia	Australia	Australia	IRD	CY080555	AACGCTGTGCCTTGGGCACCATGCAGTACCAAACGGAACGATAGTGAAAACAATCACGAATGACCAAATTGAAGTTACTA
A/Austria/211557/2005	2005-03-01	2005.162	day	Europe	Austria	Austria	Austria	IRD	EU106666	AACGCTGTGCCTTGGGCACCATGCAGTACCAAACGGAACGATAGTGAAAACAATCACGAATGACCAAATTGAAGTTACTA
A/Bahia/100/2011	2011-03-24	2011.225	day	SouthAmerica	Brazil	Brazil	Bahia	IRD	CY099978	AACGCTGTGCCTTGGGCACCATGCAGTGCCAAACGGAACGATAGTGAAAACAATCACGAATGACCAAATTGAAGTTACTA
A/Bangladesh/C5_12/2003	2003-XX-XX	2003.0	year	India	Bangladesh	Bangladesh	Bangladesh	IRD	DQ179512	AACGCTGTGCCTTGGGCACCATGCAGTACCAAACGGAACGATAGTGAAAACAATCACGAATGACCAAATTGAAGTTACTA
A/Beijing/12611/2010	2010-11-01	2010.833	day	NorthChina	China	Beijing	Beijing	Russell	NA	AACGCTGTGCCTTGGGCACCATGCAGTACCAAACGGAACGATAGTGAAAACAATCACGAATGACCAAATTGAAGTTACTA	
...
					</code></pre>
				</section>
				
				<section>
					<h2>Step 3. <span class="fragment strike-through">?????</span></h2>
				</section>
				
				<section>
					<h2>Step 3. Process data to generate summary statistics</h2>
					<a href="http://bedford.io/projects/dynamics-practical/">Use phylogenetics to infer evolutionary relationships.</a>						
					<pre><code data-trim>
7 2664 2668 2669 2685 2747 2752 2753 2755 2756 2744 2745 2741 2739 2727 2634 180 77 78 254 303 2621 2606 2617 320 343 356 373
0 8 9 10 244 304 305 306 315 357 390 400 419 457 458 504 535 573 593 607 623 645 868 869 871 872 873 874 878 934 935 936 937 
3->0 7->3 8->0 2642->8 2643->2642 2644->2643 2656->2644 2657->2656 2662->2657 2664->2662 2666->2656 2667->2666 2668->2667 266
0->India 3->Europe 7->SoutheastAsia 8->USACanada 2642->Europe 2643->USACanada 2644->Europe 2656->USACanada 2657->Europe 2662
0->{1997.685450,5.683161} 3->{1999.233555,0.000000} 7->{2000.085000,0.000000} 8->{1998.544642,11.366322} 2642->{1999.114123,
7->A/BANGKOK/62/2000 2664->A/BANGKOK/86/2001 2668->A/BANGKOK/178/2000 2669->A/BANGKOK/202/2000 2685->A/SINGAPORE/2/2000 2747
0->{0.000000,0.000000} 3->{0.000000,0.000000} 7->{0.000000,0.000000} 8->{0.000000,0.000000} 2642->{0.000000,0.000000} 2643->
					</code></pre>					
				</section>
				
				<section>
					<h2>Step 4. <span class="fragment strike-through">PROFIT!!!</span></h2>
				</section>
				
				<section>
					<h2>Step 4. Make cool figures</h2>
					<img class="stretch" src="images/h3_map.png">
				</section>																		
				
				<section>
					<h3>Flu phylogeny shows how viruses are related to one another</h3>
					<img class="stretch" src="images/h3_tree.png">
				</section>	
				
				<section>
					<h3>USA viruses tend to be evolutionary dead-ends</h3>
					<img class="stretch" src="images/h3_usatree.png">
				</section>		
				
				<section>
					<h3>Chinese viruses tend to be evolutionary starting points</h3>
					<img class="stretch" src="images/h3_schinatree.png">
				</section>	
				
				<section>
					<h3>Viruses trace their origins primary to China, India and Southeast Asia</h3>
					<img class="stretch" src="images/h3_trunkpro.png">
				</section>
				
				<section>
					<h2>If data is looked at in the right way, answers become immediately clear</h2>
				</section>																						
																
				<section>
					<h3><a href="http://www.edwardtufte.com/">Edward Tufte</a></h3>
					<a href="http://www.amazon.com/The-Visual-Display-Quantitative-Information/dp/0961392142">
						<img class="fragment fade-out" data-fragment-index="1" width="20%" src="images/visual-display.jpg">
					</a>
					<a href="http://www.amazon.com/Envisioning-Information-Edward-R-Tufte/dp/0961392118">
						<img data-fragment-index="1" width="20%" src="images/envisioning-information.jpg">
					</a>
					<a href="http://www.amazon.com/Visual-Explanations-Quantities-Evidence-Narrative/dp/0961392126">
						<img data-fragment-index="1" width="20%" src="images/visual-explanations.jpg">
					</a>
					<a href="http://www.amazon.com/Beautiful-Evidence-Edward-R-Tufte/dp/0961392177">
						<img class="fragment fade-out" data-fragment-index="1" width="20%" src="images/beautiful-evidence.jpg">
					</a>
				</section>		
				
				<section>
					<h2>Above all, show the data</h2>
					A good graphic is <i>honest</i>
				</section>		
				
				<section>
					<h3>Barchart of average value</h3>
					<img class="stretch" src="images/barchart.png">
				</section>		
				
				<section>
					<h3>Box-and-whisker plot showing variation</h3>
					<img class="stretch" src="images/barchart_whiskers.png">
				</section>		
				
				<section>
					<h3>Showing the data</h3>
					<img class="stretch" src="images/barchart_points.png">
				</section>																
								
				<section>
					<h3><a href="http://www.edwardtufte.com/">To clarify add detail</a></h3>
					"To clarify, add detail... Clutter and overload are not attributes of information, they are failures of design. If the information is in chaos, don’t start throwing out information, instead fix the design." <br> – Tufte
				</section>								
				
				<section data-background="black">
					<h3><a href="http://www.pnas.org/content/early/2012/01/04/1119859109">Evolution of <i>Conus</i> shells</a></h3>
					<img class="stretch" src="images/conus.jpg">
				</section>	
				
				<section>
					<h3><a href="http://www.pnas.org/content/109/30/11944.full">Greenland rising</a></h3>
					<img class="stretch" src="images/greenland.jpg">
				</section>		
								
				<section>
					<h3><a href="http://en.wikipedia.org/wiki/Small_multiple">Small multiples</a></h3>
					<img class="stretch" src="images/vouchers.jpg">
				</section>	

				<section>
					<h2>Mapping infection spread</h2>
					<a href="http://en.wikipedia.org/wiki/John_Snow_(physician)">John Snow</a> and the founding of epidemiology
				</section>	

				<section data-background="images/snow.jpg">
				</section>	
							
				<section data-background="black">
					<h3 style="color: white;">Global pandemic spread</h3>
					<img class="stretch" src="images/pandemic.gif">
				</section>	
				
				<section>
					<h3><a href="http://www.sciencemag.org/content/342/6164/1337">Geographic distance does not describe pandemic spread</a></h3>
					<img class="stretch" src="images/pandemic_geo.png">
				</section>	
				
				<section>
					<h3><a href="http://www.sciencemag.org/content/342/6164/1337">Shortest network path captures pandemic spread</a></h3>
					<img class="stretch" src="images/pandemic_network.png">
				</section>		
				
				<section>
					<h2>Predict almost exactly when zombie invasion will arrive
					<br>
					<span class="fragment">(except zombies don't get on planes. Nvm...)</span>
					</h2>
				</section>													
						
				<section>
					<h3>Animating evolution of trees</h3>
					<a href="http://bedford.io/projects/coaltrace/coaltrace.html"><img src="images/flu_tree.png"></a>
				</section>	
				
				<section>
					<h3>Animating evolution of haplotypes</h3>
					<a href="http://bedford.io/projects/haplotypes/haplotypes.html"><img src="images/epas1_haplotype_plot.png"></a>
				</section>		
				
				<section>
					<h1><a href="http://earth.nullschool.net/#current/wind/surface/level/orthographic=-120.62,47.47,636	">earth</a></h1>
				</section>																																		
				
			<section>
					<h2>Thanks for listening</h2>
				</section>				
				
			</div>

		</div>

		<script src="lib/js/head.min.js"></script>
		<script src="js/reveal.min.js"></script>

		<script>

			// Full list of configuration options available here:
			// https://github.com/hakimel/reveal.js#configuration
			Reveal.initialize({
				controls: false,
				progress: true,
				history: true,
				center: true,
				margin: 0.05,
				minScale: 0.1,
    			maxScale: 5.0,
    			width: 960,
    			height: 720,

				theme: Reveal.getQueryHash().theme, // available themes are in /css/theme
				transition: Reveal.getQueryHash().transition || 'linear', // default/cube/page/concave/zoom/linear/fade/none

				// Parallax scrolling
				// parallaxBackgroundImage: 'https://s3.amazonaws.com/hakim-static/reveal-js/reveal-parallax-1.jpg',
				// parallaxBackgroundSize: '2100px 900px',

				// Optional libraries used to extend on reveal.js
				dependencies: [
					{ src: 'lib/js/classList.js', condition: function() { return !document.body.classList; } },
					{ src: 'plugin/markdown/marked.js', condition: function() { return !!document.querySelector( '[data-markdown]' ); } },
					{ src: 'plugin/markdown/markdown.js', condition: function() { return !!document.querySelector( '[data-markdown]' ); } },
					{ src: 'plugin/highlight/highlight.js', async: true, callback: function() { hljs.initHighlightingOnLoad(); } },
					{ src: 'plugin/zoom-js/zoom.js', async: true, condition: function() { return !!document.body.classList; } },
					{ src: 'plugin/notes/notes.js', async: true, condition: function() { return !!document.body.classList; } }
				]
			});
			
		</script>

	</body>
</html>
